ragna/deploy/_ui/main_page.py

```diff
@@ -57,10 +64,19 @@ def after_update_chats(self):
                     self.central_view.set_current_chat(c)
                     break
 
-    # Modal and callbacks
-    def open_modal(self):
+    async def open_modal(self, event):
+        if (
+            self.components is None
+            or self.corpus_metadata is None
+            or self.corpus_names is None
+        ):
+            await self.refresh_data()
```

Comment:
On UI startup, `refresh_data` is called by `asyncio.ensure_future`. It's possible (but probably unlikely) that it will not have been called by the time a user clicks on the 'New Chat' button, and so we introduce a guard here.

We also need to convert this function into an async function as a result.